how-to/web-interop-support-context-and-intents/client/src/platform/layout/layout-override.ts

```diff
@@ -0,0 +1,258 @@
+					if (layoutElement !== null) {
+						return { layoutName: layoutElement.id, uuid: fin.me.uuid, name: fin.me.name };
+					}
+				}
+
+				throw new Error("View not found in any layout");
+			}
+
+			/**
+			 * Given a layout identity it determines if the layout is currently visible.
+			 * @param layoutName The layout identity.
+			 * @param layoutName.layoutName The name of the layout.
+			 * @returns Whether the layout is visible.
+			 */
+			public isLayoutVisible({ layoutName }: OpenFin.LayoutIdentity): boolean {
+				return layoutName === this._selectedLayout;
+			}
+
+			/**
+			 * This returns the number of layouts in the layout manager.
+			 * @returns The number of layouts in the layout manager.
+			 */
+			public size(): number {
+				return this._layoutMapArray.length;
+			}
+
+			/**
+			 * Removes this layout from the state and the DOM.
+			 * @param layoutName The name of the layout to remove.
+			 * @param index The index of the layout that is being removed.
+			 */
+			private async removeThisLayout(layoutName: string, index: number): Promise<void> {
+				// remove layout from state.
+				const layoutNameElement = document.querySelector<HTMLElement>(`#${layoutName}`);
+				if (layoutNameElement) {
+					layoutNameElement.remove();
+					await fin.Platform.Layout.destroy({ layoutName, uuid: fin.me.uuid, name: fin.me.name });
+					this._layoutMapArray = this._layoutMapArray.filter((x) => x.layoutName !== layoutName);
+					const nextLayoutName =
+						this._layoutMapArray[index]?.layoutName ?? this._layoutMapArray[index - 1]?.layoutName;
+					if (this._layoutSelector !== null) {
+						for (let i = 0; i < this._layoutSelector.options.length; i++) {
+							if (this._layoutSelector.options[i].value === layoutName) {
+								this._layoutSelector.remove(i);
+								break;
+							}
+						}
+						this.bindLayoutSelector(nextLayoutName, false);
+						await this.showLayout({ layoutName: nextLayoutName, uuid: fin.me.uuid, name: fin.me.name });
+					}
+				}
+			}
+
+			/**
+			 * A Create function for layouts.
+			 * @param layoutName A string for the layout name.
+			 * @param layout LayoutOptions.
+			 * @param entry the entry from the batch that is being created.
+			 * @param length the total number of layouts to create.
+			 * @param selectedLayout the layout that is selected.
+			 */
+			private async createLayout(
+				layoutName: string,
+				layout: OpenFin.LayoutOptions,
+				entry: number,
+				length: number,
+				selectedLayout: string | undefined
+			): Promise<void> {
+				// Create a new div container for the layout.
+				const container = document.createElement("div");
+				container.id = layoutName;
+				container.className = "col fill layout-container";
+				container.style.display = "none";
+				this._layoutContainer?.append(container);
+				if (entry === length) {
+					this.bindLayoutSelector(selectedLayout ?? layoutName);
+					await this.showLayout({
```

Comment:
Curious, why show the layout before it's created?

Reply:
switched it so it creates the layout before showing.